### ISSUE TITLE
Fixes to Product CRUD

### ISSUE THREAD

**Issue by iammukeshm** [fluentpos/fluentpos#97]
- [ ] Return  Brand Name / Category Name instead of ID from server
- [ ] Upload Image
- [ ] Multi Column Form Dialog

![image](https://user-images.githubusercontent.com/31455818/127759391-545e3e14-fe13-4c0e-8628-aeacc05df6ac.png)

**Comment by chhinsras** [comment]
WIP

**Comment by iammukeshm** [comment]
- [ ] Let it return ID also,  we may need it somewhere. But make sure Names are displayed on the table.
- [ ] Add Brand / Category Selection during Update / Create Operations

@chhinsras

**Comment by chhinsras** [comment]
![image](https://user-images.githubusercontent.com/31751665/127770620-bf4dd288-881c-4b59-98de-07a722b00ce8.png)

*Added Ids back.

**Comment by iammukeshm** [comment]
Is there any way to choose the required columns on runtime? Like, I don't want to see ID, Localname, Alert columns now, but tomorrow maybe I want to see the alerts..Can this be done without code change? Generally displaying ID in this table is not required most of the time...But we cant avoid it as well.

**Comment by unchase** [comment]
> Is there any way to choose the required columns on runtime? Like, I don't want to see ID, Localname, Alert columns now, but tomorrow maybe I want to see the alerts..Can this be done without code change? Generally displaying ID in this table is not required most of the time...But we cant avoid it as well.

I propose to move this into the client settings (for each displayed entity).
The POS administrator will be able to independently choose which properties of the entity to display and which not.

**Comment by iammukeshm** [comment]
Yes, it will be client-side only. Some kind of custom column chooser attached to ng material table.

**Comment by chhinsras** [comment]
- [ ] Tax is Decimal / Boolean ? Currently . it is string. 
- [ ] Error Registering Product Due to Tax

**Comment by chhinsras** [comment]
@iammukeshm @unchase what do you plan for uploading. currently. the file-system based upload has bee done.
